src/components/AccountIconChooser.js

```diff
@@ -16,143 +16,106 @@
 
 // @flow
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {
-  FlatList,
   StyleSheet,
-  Text,
   TouchableOpacity,
   View
 } from 'react-native';
+import Icon from 'react-native-vector-icons/MaterialIcons';
 import colors from '../colors';
 import { brainWalletAddress, words } from '../util/native';
-import AccountIcon from './AccountIcon';
 
-export default class AccountIconChooser extends React.PureComponent<{
-  value: string,
-  onChange: () => any
-}> {
-  static propTypes = {
-    value: PropTypes.string,
-    onChange: PropTypes.func
-  };
+import AccountIcon from './AccountIcon';
+import Address from './Address'
 
+export default class AccountIconChooser extends React.PureComponent {
   constructor(props) {
     super(props);
 
-    this.state = { icons: [] };
+    this.state = {};
```

Comment:
IMHO, put all the state even the `null` one in the constructor will give a quick overview of the whole component's structure. For example, I am a little bit surprised when I see that line in the later code.
```javascript
const { address, bip39, seed } = this.state;
```